framework/utils/graphStructure.py

```diff
@@ -202,52 +203,56 @@ def findAllPaths(self, startVertex, endVertex, path=[]):
+  def isConnected(self, graphDict, verticesEncountered = None, startVertex=None):
     """
       Method that determines if the graph is connected (graph theory connectivity)
       @ In, verticesEncountered, set, the encountered vertices (generally it is None when called from outside)
+      @ In, graphDict, dict, the graph dictionary
       @ In, startVertex, string, the starting vertex
       @ Out, isConnected, bool, is the graph fully connected?
     """
     if verticesEncountered is None:
       verticesEncountered = set()
-    gdict = self.__graphDict
-    vertices = list(gdict.keys())
+    vertices = list(graphDict.keys())
     if not startVertex:
       # chosse a vertex from graph as a starting point
       startVertex = vertices[0]
     verticesEncountered.add(startVertex)
     if len(verticesEncountered) != len(vertices):
-      for vertex in gdict[startVertex]:
+      for vertex in graphDict[startVertex]:
         if vertex not in verticesEncountered:
-          if self.isConnected(verticesEncountered, vertex):
+          if self.isConnected(graphDict, verticesEncountered, vertex):
             return True
     else:
       return True
     return False
 
-  def isConnectedNet(self, startVertex=None):
+  def isConnectedNet(self):
     """
       Method that determines if the graph is a connected net (all the vertices are connect with each other through other vertices)
-      This method can state that we have a connected net even if, based on graph theory, the graph is not connected
-      @ In, startVertex, string, the starting vertex
+      @ In, None
       @ Out, graphNetConnected, bool, is the graph net fully connected?
     """
-    graphNetConnected = self.isConnected()
-    if graphNetConnected:
-      return graphNetConnected
-    # the graph is not connected (Graph theory)
-    uniquePaths = self.findAllUniquePaths()
-    # now check if there is at list a node that is common in all the paths
-    if len(uniquePaths) > 0:
-      graphNetConnected = True
-      startPath = set(uniquePaths.pop())
-      for otherPath in uniquePaths:
-        if startPath.isdisjoint(otherPath):
-          graphNetConnected = False
-          break
+    graphDict = self.__extendDictForGraphTheory()
+    graphNetConnected = self.isConnected(graphDict)
     return graphNetConnected
 
+  def __extendDictForGraphTheory(self):
+    """
+      Method to extend the graph dictionary in order to be accepted by the graph theory
+      @ In, None
+      @ Out, graphDict, dict, the extended graph dictionary, used for isConnectedNet and findIsolatedVertices
+    """
+    graphDict = copy.deepcopy(self.__graphDict)
+    # Extend graph dictionary generated by ensemble model to graph theory acceptable dictionary
+    # Basicly, if a -> b (a is connected to b), the self.__graphDict = {'a':['b'], 'b':[]}
+    # Since a is connected to b, from the graph theory, the dictionary should become {'a':['b'], 'b':['a']}
```

Comment:
This class was supposed to represent a ```directed graph``` since was mainly used by the ensemble model for the identification of the execution order. 
Please, put a WARNING to avoid to store this extended graph dictionary in ```__graphDict``` and specify that this should be used for the isConnected method only.

In addition, enhance  the comment above (and in the above docstrings) to state this.